### PR TITLE
fix(x/distribution): Nakamoto coefficient update change

### PR DESCRIPTION
**Pull request by giuliostramondo**
V4 Audit, issue 22

The size of the high group and low group could differ if the total number of bonded validators is not evenly divisible by 3.

Without this change the lowest group of validators might have an additional validator with respect to the highest group of validators. This would slightly increase the low group VP average and slightly bias the Nakamoto bonus coefficient towards lower numbers.

### CODE REVIEW

**Review by Pantani — APPROVED** [review]

**Comment by tbruyelle** [comment]
@giuliostramondo Please add a test.

File: x/distribution/keeper/nakamoto_bonus.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func (k Keeper) AdjustNakamotoBonusCoefficient(ctx sdk.Context) error {
 	// split into 3 groups as evenly as possible: high, medium, low
 	groupSize := n / 3
 	high := validators[:groupSize]
-	low := validators[groupSize*2:]
+	low := validators[n-groupSize:]
```
Comment (Pantani) [inline x/distribution/keeper/nakamoto_bonus.go:57]: Can we add a regression test for a validator count where `n % 3 != 0` here (for example, 5 validators)? This change relies on high and low having the same size, and we don’t currently have coverage for that audit edge case

**Review by julienrbrt — APPROVED** [review]
